frb_rust/src/ffi/io.rs

```diff
@@ -31,3 +40,63 @@ mod tests {
         assert_eq!(ns, 567_123_000);
     }
 }
+
+#[derive(Debug)]
+pub struct DartOpaque {
+    /// Dart object handle that won't be removed by GC.
+    handle: Dart_PersistentHandle,
+
+    /// Used to drop the `handle` on the Dart thread
+    port: Channel,
+
+    /// Flag to cancel the drop when sending to dart.
+    drop: bool,
+
+    /// The ID of the thread on which the Dart Object was created.
+    id: ThreadId,
```

Comment:
uber-nit: what about thread_id, since `DartOpaque.id` looks like "id of the object"